redash/handlers/query_results.py

```diff
@@ -171,11 +171,15 @@ def post(self, query_id):
         max_age = int(max_age)
 
         query = get_object_or_404(models.Query.get_by_id_and_org, query_id, self.current_org)
-
         allow_executing_with_view_only_permissions = query.parameterized.is_safe
 
         if has_access(query, self.current_user, allow_executing_with_view_only_permissions):
-            return run_query(query.parameterized, parameter_values, query.data_source, query_id, max_age)
+            query_result = run_query(query.parameterized, parameter_values, query.data_source, query_id, max_age)
+            if isinstance(query_result, tuple) and query_result[1] != 200:
```

Comment:
As this is not the only place we return a `QueryResult` instance, this behavior needs to be in a serializer. I don't think we created one for a `QueryResult` yet and we still use a `to_dict` method, but we can as part of this change.